### ISSUE TITLE
Structured data always show Out of Stock for variable products

### ISSUE THREAD

**Issue by leocheungswell** [woocommerce/woocommerce#21504]
**Is your feature request related to a problem? Please describe.**
Kind of. Google search listing always showing variable sku "Out of Stock"

Google Listing:
Product Name - Domain
https://www.domain.com › Collections › Product › Item
$25.00 to $45.00 - ‎Out of stock

**Describe the solution you'd like**
We have turn off the inventory management for Variable SKU and let the variation handles the product stock. But Google picks up the availability from the parent product not the variation which could hurt conversion.

/woocommerce/includes/class-wc-structured-data.php

			$markup_offer += array(
				'priceCurrency' => $currency,
				'availability'  => 'https://schema.org/' . ( $product->is_in_stock() ? 'InStock' : 'OutOfStock' ),
				'url'           => $markup['@id'],
				'seller'        => array(
					'@type' => 'Organization',
					'name'  => $shop_name,
					'url'   => $shop_url,
				),
			);

**Describe alternatives you've considered**
We're currently removing the product availability via the filter.

**Additional context**
If we can somehow show the variations stock level on the admin product listing page that will be perfect. We always have to click inside the product in order to figure out the current stock level and it's always saying out of stock on the listing view.

**Comment by woocommercebot** [comment]
Thanks for the suggestion @leocheungswell.

We'll keep an eye on the [popularity of this request](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md#feature-requests) :)

**Comment by dmchale** [comment]
I added my +1 for this as well, just discovered it this morning. This seems to affect much more than just the structured data though... the root of the issue seems to be due to `$product->is_in_stock()` not checking its variations for product stock at all. On one of my client sites I found that all variable products (which handle stock at the variable level) ALWAYS return `False` for this function call, even if all of its variations are in stock.
